files/00_setup_freeipa.sh

```diff
@@ -151,11 +151,18 @@ function setup {
     /var/kerberos/krb5kdc/kdc.conf
   systemctl restart krb5kdc.service
 
+  # Grab the instance ID from the AWS Instance Meta-Data Service
+  # (IMDSv2)
+  imds_token=$(curl --silent \
+      --request PUT \
+      --header "X-aws-ec2-metadata-token-ttl-seconds: 10" \
+    http://169.254.169.254/latest/api/token)
+  instance_id=$(curl --silent \
+      --header "X-aws-ec2-metadata-token: $imds_token" \
+    http://169.254.169.254/latest/meta-data/instance-id)
```

Comment:
Is it worthwhile to add a check to ensure that the `instance_id` is valid (or at least not empty) before we attempt to use it?

Reply:
What should we do if it doesn't match an appropriate regex, say?

Reply:
If we have confidence in our regex, then I say we print a useful error message and bail out.

Reply:
I don't think it makes sense to bail out, since by that point we have already set up the FreeIPA server and have completed 95% of the script, but I can print an error message stating that the attempt to create the alternate hostname in FreeIPA has succeeded.

Reply:
Agreed- that makes more sense.  I thought (without checking) that this code was at the beginning of the script, not so close to the end.

Reply:
See commit edd4679.